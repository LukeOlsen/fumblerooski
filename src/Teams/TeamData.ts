import { ppaQueryResults, ppaApi, spDataPacket } from "./TeamInterfaces";
import { Utilities } from "../tools/Utilities";

export default class TeamFunctions {
  public static async cleanPPA(
    data: Array<ppaQueryResults>
  ): Promise<ppaApi[]> {
    let result: Array<ppaApi> = [];
    let offense: ppaApi = {
      id: "offense",
      color: "hsl(337, 70%, 50%)",
      data: [],
    };
    data.forEach((a: ppaQueryResults) => {
      offense.data.push({
        id: `${a.opponent}`,
        Offense: a.offense_overall,
        Defense: a.defense_overall,
      });
    });
    result.push(offense);
    return result;
  }

  public static async cleanSpRank(data: object): Promise<spDataPacket[]> {
    const propNames: Array<string> = Object.keys(data);
    let cleanedData: Array<spDataPacket> = [];
    propNames.forEach(async (el) => {
      let tempStr = await Utilities.Strings.CleanCamelString(el);
      cleanedData.push({
        subject: tempStr,
        dataSet: data[el],
        fullMark: 150,
      });
    });
    return cleanedData;
  }
}
