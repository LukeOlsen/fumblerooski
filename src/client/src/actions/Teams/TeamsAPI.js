import axios from "axios";
import { setTeamData, doneLoading, setTeamSPRank } from "../index";

export function getTeamData(team, year) {
  return (dispatch, getState) => {
    axios.get(`/api/team/teamData/${team}/${year}`).then((res) => {
      dispatch(setTeamData(res.data));
      dispatch(doneLoading());
    });
  };
}

export function getTeamSPRanking(team, year) {
  console.log(team, year);
  return (dispatch) => {
    axios.get(`/api/team/teamSPRank/${team}/${year}`).then((res) => {
      console.log(res);
      dispatch(setTeamSPRank(res.data));
    });
  };
}
