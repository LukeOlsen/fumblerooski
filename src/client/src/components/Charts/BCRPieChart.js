import React from "react";
import {
  PieChart,
  Pie,
  Legend,
  Tooltip,
  ResponsiveContainer,
  Cell,
} from "recharts";

const colors = ["#66a1ff", "#82ca9d"];

const BCRPie = (props) => {
  console.log(props.data);
  if (props.data.length > 1) {
    return (
      <ResponsiveContainer>
        <PieChart>
          <Pie data={props.data} fill="#8884d8" label>
            {props.data.map((entry, index) => {
              return <Cell key={`cell-${index}`} fill={colors[index]} />;
            })}
          </Pie>
          <Tooltip />
        </PieChart>
      </ResponsiveContainer>
    );
  } else {
    return null;
  }
};

export default BCRPie;
