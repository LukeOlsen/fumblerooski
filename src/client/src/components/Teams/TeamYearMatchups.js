import React from "react";

const SimpleMatchup = ({ props }) => {
  return (
    <div className="w-full overflow-auto h-full">
      {props.map((el) => {
        return (
          <div className="flex text-center justify-center">
            <div className="w-40 m-2 text-left">
              <div>{el.home_team}</div>
            </div>
            <div className="w-8 m-2">
              <div>{el.home_points}</div>
            </div>
            <div className="w-5 m-2">
              <div>-</div>
            </div>
            <div className="w-8 m-2">
              <div>{el.away_points}</div>
            </div>
            <div className="w-40 m-2 text-right">
              <div>{el.away_team}</div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default SimpleMatchup;
