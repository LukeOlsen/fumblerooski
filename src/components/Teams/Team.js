import React, { Component } from 'react';
import { connect } from "react-redux";

const mapStateToProps = state => {
    return {
        team: state.teamReducer.team,
        data: state.teamReducer.data
    }
}


class Team extends Component {
    constructor(props) {
        super(props)

        console.log(this.props)
    }


    render() {
        return (
            <div>
                <div>
                    <h1 className="mt-8 text-6xl">Welcome to your team</h1>
                    <h2 className="text-6xl mt-8">{this.props.team}</h2>
                    {/* <p>{this.props.data}</p> */}
                </div>
            </div>
        )
    }
}

export default connect(mapStateToProps)(Team);