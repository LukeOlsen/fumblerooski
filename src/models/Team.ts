import {
  Table,
  Column,
  Model,
  HasMany,
  PrimaryKey,
  DataType,
  BelongsTo,
  HasOne,
  ForeignKey,
} from "sequelize-typescript";
import Games from "./Games";
import Records from "./Records";
import Recruits from "./Recruits";
import PPAGameAverages from "./PPAGameAverages";
import BetLines from "./BetLines";
import { SRSRankings } from "./SRSRankings";
import { SPRankings } from "./SPRankings";

@Table
export default class SchoolsFBS extends Model<SchoolsFBS> {
  @PrimaryKey
  @Column
  id: number;

  @Column
  school: string;

  @Column
  mascot: string;

  @Column
  abbreviation: string;

  @Column
  alt_name1: string;

  @Column
  alt_name2: string;

  @Column
  alt_name3: string;

  @Column
  conference: string;

  @Column
  division: string;

  @Column
  color: string;

  @Column
  alt_color: string;

  @Column
  logos_1: string;

  @Column
  logos_2: string;

  @HasMany(() => Games, "home_id")
  homeGames: Games[];

  @HasMany(() => Games, "away_id")
  awayGames: Games[];

  @HasMany(() => Records, "team_id")
  teamRecord: Records[];

  @HasMany(() => PPAGameAverages, "team_id")
  ppaGameAverages: PPAGameAverages[];

  @HasMany(() => Recruits, "team_id")
  recruits: Recruits[];

  @HasMany(() => BetLines, "home_id")
  homeBet: BetLines[];

  @HasMany(() => BetLines, "away_id")
  awayBet: BetLines[];

  @HasMany(() => SRSRankings, "away_id")
  srs: SRSRankings[];

  @HasMany(() => SPRankings, "team_id")
  spRank: SRSRankings[];
}
