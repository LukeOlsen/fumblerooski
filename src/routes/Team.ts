import { Router } from "express";
import { Op } from "sequelize";
import SchoolsFBS from "../models/Team";
import Recruits from "../models/Recruits";
import Talents from "../models/Talents";
import Games from "../models/Games";
import Records from "../models/Records";
import PPAGameAverages from "../models/PPAGameAverages";
import TeamFunctions from "../Teams/TeamData";

export const team = Router();

team.get("/", (req, res) => {
  SchoolsFBS.findAll().then((teams) => {
    res.send(teams);
  });
});

team.get("/teamData/:team/:year", async (req, res, next) => {
  const schoolData = SchoolsFBS.findAll({
    where: {
      school: req.params.team,
    },
    include: [
      {
        model: Games,
        as: "homeGames",
        where: {
          [Op.or]: [
            { home_team: req.params.team },
            { away_team: req.params.team },
          ],
          season: req.params.year,
        },
      },
      {
        model: Games,
        as: "awayGames",
        where: {
          [Op.or]: [
            { home_team: req.params.team },
            { away_team: req.params.team },
          ],
          season: req.params.year,
        },
      },
      {
        model: Records,
        as: "teamRecord",
        where: {
          year: req.params.year,
          team: req.params.team,
        },
      },
      {
        model: PPAGameAverages,
        attributes: [
          "week",
          "season",
          "opponent",
          "offense_firstDown",
          "offense_overall",
          "offense_passing",
          "offense_rushing",
          "offense_secondDown",
          "offense_thirdDown",
          "defense_firstDown",
          "defense_overall",
          "defense_passing",
          "defense_rushing",
          "defense_secondDown",
          "defense_thirdDown",
        ],
        where: {
          team: req.params.team,
          season: req.params.year,
        },
      },
    ],
  });

  const recruitData = Recruits.findAll({
    where: {
      committedTo: req.params.team,
      year: req.params.year,
    },
  });

  const talentData = Talents.findAll({
    where: {
      school: req.params.team,
      year: req.params.year,
    },
  });

  Promise.all([schoolData, recruitData, talentData])
    .then(async (result: any) => {
      let newPpaFormat = await TeamFunctions.cleanPPA(
        result[0][0].ppaGameAverages
      );
      result.push(newPpaFormat);
      res.send(result);
    })
    .catch((err) => {
      console.log(err);
    });
});

// var express = require("express");
// var teamData = express.Router();
// var teamQ = require("./TeamQueries");

// teamData.get("/:team", async (req, res) => {
//   let teamData = await teamQ(req.params.team);
//   res.json(teamData);
// });
