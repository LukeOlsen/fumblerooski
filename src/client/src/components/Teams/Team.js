import React, { Component } from "react";
import { connect } from "react-redux";
import { getTeamData } from "../../actions/Teams/TeamsAPI";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAngleDown } from "@fortawesome/free-solid-svg-icons";
import RecruitsTable from "../Recruits/RecruitsTable";
import PPALine from "../Charts/PPALineChart";
import BCRPieChart from "../Charts/BCRPieChart";

const mapStateToProps = (state) => {
  console.log(state);
  return {
    team: state.teamReducer.team,
    data: state.teamReducer.data,
    teamInfo: state.teamReducer.teamInfo,
    recruits: state.teamReducer.latestRecruits,
    talent: state.teamReducer.latestTalentRating,
    PPA: state.teamReducer.ppaAverages,
    BCR: state.teamReducer.BCR,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    // setTeam: team => dispatch(setTeam(team)),
    getTeamData: (team, year) => dispatch(getTeamData(team, year)),
  };
};

let currentTeam = "";

const RenderLogo = (props) => {
  console.log(props);
  if (props) {
    return (
      <div className="w-full md:flex-1 self-center">
        <img className="h-40 object-center" src={props.logos_1} alt="" />
      </div>
    );
  } else {
    return null;
  }
};

class Team extends Component {
  constructor(props) {
    super(props);

    this.state = {
      year: 2019,
    };

    this.handleYearChange = this.handleYearChange.bind(this);
    this.handleYearSubmit = this.handleYearSubmit.bind(this);
  }

  handleYearChange(event) {
    this.setState({ year: event.target.value });
    this.props.getTeamData(
      this.props.match.params.teamName,
      event.target.value
    );
  }

  handleYearSubmit() {
    this.props.getTeamData(this.props.match.params.teamName, this.state.year);
  }

  componentDidMount() {
    let year = this.state.year;
    this.props.getTeamData(this.props.match.params.teamName, this.state.year);
    currentTeam = this.props.match.params.teamName;
  }

  componentDidUpdate() {
    console.log(this.state.year);
    if (this.props.match.params.teamName != currentTeam) {
      this.props.getTeamData(this.props.match.params.teamName, this.state.year);
      currentTeam = this.props.match.params.teamName;
    }
  }

  render() {
    return (
      <div className="p-2 pl-10 flex-grow max-h-full overflow-y-scroll flex-row min-h-screen">
        <div className="flex">
          <div className="flex-auto">
            <p className="text-4xl">
              {this.props.match.params.teamName}{" "}
              {this.props.teamInfo[0] ? this.props.teamInfo[0].mascot : ""}
            </p>
          </div>
          <div className="flex-auto flex text-center">
            <form
              className="flex-1 self-center"
              onSubmit={this.handleYearSubmit}
            >
              <div className="inline-block relative w-20 rounded">
                <select
                  className="block leading-tight text-black mr-10 w-30 appearance-none py-1 px-2 pr-12 rounded"
                  value={this.state.teamYear}
                  onChange={this.handleYearChange}
                >
                  <option>2019</option>
                  <option>2018</option>
                  <option>2017</option>
                  <option>2016</option>
                  <option>2015</option>
                  <option>2014</option>
                  <option>2013</option>
                  <option>2012</option>
                  <option>2011</option>
                  <option>2010</option>
                  <option>2009</option>
                  <option>2008</option>
                  <option>2007</option>
                  <option>2006</option>
                  <option>2005</option>
                  <option>2004</option>
                  <option>2003</option>
                  <option>2002</option>
                  <option>2001</option>
                  <option>2000</option>
                </select>
                <div className="pointer-events-none absolute inset-y-0 right-0 left-2 flex items-center text-gray-700">
                  <FontAwesomeIcon icon={faAngleDown} />
                </div>
              </div>
            </form>
            <div className="flex-1 text-2xl self-center">
              {this.props.teamInfo[0] ? this.props.teamInfo[0].conference : ""}
            </div>
            <div className="flex-1 text-2xl self-center">
              {this.props.teamInfo[0] ? this.props.teamInfo[0].division : ""}
            </div>
            <div className="flex-1 text-2xl self-center">
              {this.props.teamInfo[0]
                ? this.props.teamInfo[0].teamRecord[0].total_wins +
                  "-" +
                  this.props.teamInfo[0].teamRecord[0].total_losses
                : ""}
            </div>
          </div>
        </div>
        <div className="flex flex-wrap">
          <div className="bg-gray-700 rounded border-black h-64 w-full sm:flex-1 md:flex-1/3 m-2 shadow-lg">
            {this.props.PPA ? <PPALine PPA={this.props.PPA} /> : null}
          </div>
          <div className="bg-gray-700 rounded border-black h-64 w-full sm:flex-1 md:flex-1/3 m-2 shadow-lg">
            {this.props.BCR ? <BCRPieChart data={this.props.BCR} /> : null}
          </div>
        </div>
        <div className="bg-gray-700 rounded m-2">
          <RecruitsTable recruits={[this.props.recruits]} />
        </div>
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Team);
