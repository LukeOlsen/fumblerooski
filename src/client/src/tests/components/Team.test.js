import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Team } from "../../components/Teams/Team";
import { MemoryRouter } from "react-router";
import BCRPieChart from "../../components/Charts/BCRPieChart";

jest.mock("../../components/Charts/BCRPieChart", () => () => "BCRPieChart");
jest.mock("../../components/Charts/PPALineChart", () => () => "PPALineChart");
jest.mock("../../components/Recruits/RecruitsTable", () => () =>
  "RecruitsTable"
);

describe("Teams", () => {
  it("snapshot renders after receiving props and loading is set to false", () => {
    const component = renderer.create(
      <Team
        loading={false}
        match={{ params: { team: "Team" }, isExact: true, path: "", url: "" }}
        teamInfo={[
          {
            mascot: "animal",
            teamRecord: [{ total_wins: 11, total_losses: 11 }],
          },
        ]}
        getTeamData={jest.fn()}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("snapshot renders loading state when loading props is true", () => {
    const component = renderer.create(
      <Team
        loading={true}
        match={{ params: { team: "Team" }, isExact: true, path: "", url: "" }}
        getTeamData={jest.fn()}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
