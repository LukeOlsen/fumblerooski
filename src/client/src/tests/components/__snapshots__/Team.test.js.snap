// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams snapshot renders after receiving props and loading is set to false 1`] = `
<div
  className="p-2 pl-10 flex-grow max-h-full overflow-y-scroll flex-row min-h-screen"
>
  <div
    className="flex"
  >
    <div
      className="flex-auto"
    >
      <p
        className="text-4xl"
      >
         
        animal
      </p>
    </div>
    <div
      className="flex-auto flex text-center"
    >
      <form
        className="flex-1 self-center"
        onSubmit={[Function]}
      >
        <div
          className="inline-block relative w-20 rounded"
        >
          <select
            className="block leading-tight text-black mr-10 w-30 appearance-none py-1 px-2 pr-12 rounded"
            onChange={[Function]}
          >
            <option>
              2019
            </option>
            <option>
              2018
            </option>
            <option>
              2017
            </option>
            <option>
              2016
            </option>
            <option>
              2015
            </option>
            <option>
              2014
            </option>
            <option>
              2013
            </option>
            <option>
              2012
            </option>
            <option>
              2011
            </option>
            <option>
              2010
            </option>
            <option>
              2009
            </option>
            <option>
              2008
            </option>
            <option>
              2007
            </option>
            <option>
              2006
            </option>
            <option>
              2005
            </option>
            <option>
              2004
            </option>
            <option>
              2003
            </option>
            <option>
              2002
            </option>
            <option>
              2001
            </option>
            <option>
              2000
            </option>
          </select>
          <div
            className="pointer-events-none absolute inset-y-0 right-0 left-2 flex items-center text-gray-700"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-angle-down fa-w-10 "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </form>
      <div
        className="flex-1 text-2xl self-center"
      />
      <div
        className="flex-1 text-2xl self-center"
      />
      <div
        className="flex-1 text-2xl self-center"
      >
        11-11
      </div>
    </div>
  </div>
  <div
    className="flex flex-wrap flex-col md:flex-row"
  >
    <div
      className="bg-gray-700 rounded border-black h-64 w-full sm:flex-1 md:flex-1/3 m-2 shadow-lg"
    >
      PPALineChart
    </div>
    <div
      className="bg-gray-700 rounded border-black h-64 w-full sm:flex-1 md:flex-1/3 m-2 shadow-lg"
    >
      BCRPieChart
    </div>
  </div>
  <div
    className="bg-gray-700 rounded m-2"
  >
    RecruitsTable
  </div>
</div>
`;

exports[`Teams snapshot renders loading state when loading props is true 1`] = `
<div
  className="p-2 pl-10 flex-grow max-h-full overflow-y-scroll flex-row min-h-screen"
>
  <div
    className="flex"
  >
    <div
      className="flex-auto"
    >
      <p
        className="text-4xl"
      >
         
        animal
      </p>
    </div>
    <div
      className="flex-auto flex text-center"
    >
      <form
        className="flex-1 self-center"
        onSubmit={[Function]}
      >
        <div
          className="inline-block relative w-20 rounded"
        >
          <select
            className="block leading-tight text-black mr-10 w-30 appearance-none py-1 px-2 pr-12 rounded"
            onChange={[Function]}
          >
            <option>
              2019
            </option>
            <option>
              2018
            </option>
            <option>
              2017
            </option>
            <option>
              2016
            </option>
            <option>
              2015
            </option>
            <option>
              2014
            </option>
            <option>
              2013
            </option>
            <option>
              2012
            </option>
            <option>
              2011
            </option>
            <option>
              2010
            </option>
            <option>
              2009
            </option>
            <option>
              2008
            </option>
            <option>
              2007
            </option>
            <option>
              2006
            </option>
            <option>
              2005
            </option>
            <option>
              2004
            </option>
            <option>
              2003
            </option>
            <option>
              2002
            </option>
            <option>
              2001
            </option>
            <option>
              2000
            </option>
          </select>
          <div
            className="pointer-events-none absolute inset-y-0 right-0 left-2 flex items-center text-gray-700"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-angle-down fa-w-10 "
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </form>
      <div
        className="flex-1 text-2xl self-center"
      />
      <div
        className="flex-1 text-2xl self-center"
      />
      <div
        className="flex-1 text-2xl self-center"
      >
        11-11
      </div>
    </div>
  </div>
  <div
    className="flex flex-wrap flex-col md:flex-row"
  >
    <div
      className="bg-gray-700 rounded border-black h-64 w-full sm:flex-1 md:flex-1/3 m-2 shadow-lg"
    >
      PPALineChart
    </div>
    <div
      className="bg-gray-700 rounded border-black h-64 w-full sm:flex-1 md:flex-1/3 m-2 shadow-lg"
    >
      BCRPieChart
    </div>
  </div>
  <div
    className="bg-gray-700 rounded m-2"
  >
    RecruitsTable
  </div>
</div>
`;
